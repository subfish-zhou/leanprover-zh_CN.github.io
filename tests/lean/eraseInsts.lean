structure Foo where
  a : Nat

instance fooAdd : Add Foo where
  add x y := ⟨x.a + y.a⟩

def f1 (a b : Foo) := a + b

section
attribute [-instance] fooAdd

def f2 (a b : Foo) := a + b -- Error
end

def f3 (a b : Foo) := a + b
