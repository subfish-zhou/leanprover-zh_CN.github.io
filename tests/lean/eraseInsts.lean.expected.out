eraseInsts.lean:12:22-12:27: error: failed to synthesize instance
  HAdd Foo Foo ?m
